braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -167,8 +169,16 @@ private BrazeIntegrationFactory(Object config, RudderClient client, RudderConfig
         } else if (RudderClient.getApplication() == null) {
             RudderLogger.logError("RudderClient is not initialized correctly. Application is null. Aborting Braze initialization.");
         } else {
-            // get apiKey and return if null or blank
-            if (destinationConfig.containsKey(API_KEY)) {
+            // Prefer platform-specific key if present and not empty
+            if (getBoolean(destinationConfig.get(USE_PLATFORM_SPECIFIC_API_KEYS))
+                    && destinationConfig.containsKey(ANDROID_API_KEY)) {
+                apiKey = (String) destinationConfig.get(ANDROID_API_KEY);
+            }
+            // Fallback to default app key if either platform-specific key is not present or is set to false
+            if (TextUtils.isEmpty(apiKey) && destinationConfig.containsKey(API_KEY)) {
+                if (getBoolean(destinationConfig.get(USE_PLATFORM_SPECIFIC_API_KEYS))) {
+                    RudderLogger.logWarn("BrazeIntegration: usePlatformSpecificApiKeys is enabled but androidApiKey is not configured. Falling back to legacy apiKey.");
```

Comment:
Let's also include the Android API key in the message.

```suggestion
                    RudderLogger.logWarn("BrazeIntegration: Configured to use platform-specific API keys but Android API key is not valid. Falling back to the default API key.");
```